core/src/main/java/io/lionweb/utils/ChunkValidator.java

```diff
@@ -0,0 +1,115 @@
+package io.lionweb.utils;
+
+import io.lionweb.serialization.data.*;
+import java.util.*;
+import javax.annotation.Nonnull;
+
+/**
+ * Validate a generic chunk, which may contain part of a partition, an entire partition, or perhaps
+ * multiple partitions.
+ */
+public class ChunkValidator extends Validator<SerializedChunk> {
+  @Override
+  public ValidationResult validate(@Nonnull SerializedChunk chunk) {
+    Objects.requireNonNull(chunk, "chunk should not be null");
+    ValidationResult validationResult = new ValidationResult();
+
+    // Prepare supporting structures
+    Map<String, SerializedClassifierInstance> nodesByID = new HashMap<>();
+    for (SerializedClassifierInstance node : chunk.getClassifierInstances()) {
+      // Verifying IDs are valid
+      if (!CommonChecks.isValidID(node.getID())) {
+        validationResult.addError("Invalid node id: " + node.getID(), node.getID());
+      }
+
+      if (nodesByID.containsKey(node.getID())) {
+        // Verifying there are no duplicate node ids
+        validationResult.addError("Duplicate node id: " + node.getID(), node.getID());
+      } else {
+        nodesByID.put(node.getID(), node);
+      }
+    }
+
+    // Check languages
+    Set<UsedLanguage> usedLanguages = new HashSet<>();
+    for (SerializedClassifierInstance node : chunk.getClassifierInstances()) {
+      usedLanguages.add(node.getClassifier().getUsedLanguage());
+      for (SerializedPropertyValue propertyValue : node.getProperties()) {
+        usedLanguages.add(propertyValue.getMetaPointer().getUsedLanguage());
+      }
+      for (SerializedReferenceValue referenceValue : node.getReferences()) {
+        usedLanguages.add(referenceValue.getMetaPointer().getUsedLanguage());
+      }
+      for (SerializedContainmentValue containmentValue : node.getContainments()) {
+        usedLanguages.add(containmentValue.getMetaPointer().getUsedLanguage());
+      }
+    }
+    if (!usedLanguages.equals(new HashSet<>(chunk.getLanguages()))) {
+      validationResult.addError(
+          "We expected these used languages: "
+              + usedLanguages
+              + " and we found "
+              + chunk.getLanguages());
+    }
+
+    // Ensuring that containments and annotations are the inverse of parent relationships
+    Set<String> containedNodes = new HashSet<>();
+    for (SerializedClassifierInstance node : chunk.getClassifierInstances()) {
+      for (SerializedContainmentValue containmentValue : node.getContainments()) {
+        for (String childId : containmentValue.getValue()) {
+          // Verifying nodes do not appear in multiple containments or annotations
+          if (containedNodes.contains(childId)) {
+            validationResult.addError(childId + " is listed in multiple places", childId);
```

Comment:
Wouldn't it be useful to list those (multiple) places? (You could use a `Map<LionWebId, LionWebId[]>` mapping ID of contained node &rarr; IDs of container nodes.)